### PR TITLE
[rllib][minor] clarification of the softmax axis in dqn_torch_policy

### PR DESCRIPTION
**Pull request by AnnaKosiorek**
pytorch nn.functional.softmax (unlike tf.nn.softmax) calculates softmax along zeroth dimension by default

## Why are these changes needed?

softmax was calculated along batch, not along logits

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sven1977 — APPROVED** [review]
Looks great, thanks for the PR @AnnaKosiorek !

**Comment by richardliaw** [comment]
@AnnaKosiorek hey, could you help merge master back into your branch? 

sorry for the delay!

**Comment by AnnaKosiorek** [comment]
@richardliaw done
